render/egl.c

```diff
@@ -342,7 +346,7 @@ bool wlr_egl_swap_buffers(struct wlr_egl *egl, EGLSurface surface,
 		int nrects;
 		pixman_box32_t *rects =
 			pixman_region32_rectangles(&flipped_damage, &nrects);
-		EGLint egl_damage[4 * nrects];
+		EGLint *egl_damage = malloc(sizeof(EGLint) * (4 * nrects));
```

Comment:
This isn't freed?
Also per-frame or per-damage malloc is a terrible terrible idea. Don't. Ever.
Make an array somewhere in the main context and increase its size only if needed.

Reply:
Yep, thanks for spotting it. I'm not really looking at the greater context of this code so I had no idea that this was a hot code path. I'll look into implementing what you suggested.